File: MyAnimeListJava/src/main/java/com/example/myanimelistjava/models/Review.java
package com.example.myanimelistjava.models;

import lombok.Data;

import java.util.UUID;

@Data
public class Review {
    public final UUID id;
    public final UUID anime;
    public final UUID user;
    public int score;
    public String comment;


    public Review(UUID id, UUID anime, UUID user, int score, String comment) {
        this.id= id;
        this.anime = anime;
        this.user = user;
        this.score = score;
        this.comment = comment;
    }
}

File: MyAnimeList/src/main/kotlin/com/example/myanimelist/models/Review.kt
package com.example.myanimelist.models

import java.util.*

data class Review(
    val anime: Anime,
    val user: User,
    var score: Int,
    var comment: String,
    val id: UUID = UUID.randomUUID()
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Review

        if (id != other.id) return false

        return true
    }

    override fun hashCode(): Int {
        return id.hashCode()
    }
}